src/test/java/seedu/sudohr/model/leave/LeaveTest.java

```diff
@@ -0,0 +1,93 @@
+package seedu.sudohr.model.leave;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.sudohr.logic.commands.CommandTestUtil.VALID_LEAVE_DATE_LEAVE_TYPE_2;
+import static seedu.sudohr.testutil.Assert.assertThrows;
+import static seedu.sudohr.testutil.TypicalLeave.LEAVE_TYPE_1;
+import static seedu.sudohr.testutil.TypicalLeave.LEAVE_TYPE_2;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.sudohr.model.employee.exceptions.DuplicateEmployeeException;
+import seedu.sudohr.testutil.EmployeeBuilder;
+import seedu.sudohr.testutil.LeaveBuilder;
+import seedu.sudohr.testutil.TypicalEmployees;
+
+public class LeaveTest {
+
+    @Test
+    public void asObservableList_modifyList_throwsUnsupportedOperationException() {
+        Leave department = new LeaveBuilder().build();
+        assertThrows(UnsupportedOperationException.class, () -> department.getEmployees().remove(0));
+    }
+
+    @Test
+    public void execute_addEmployee_success() {
+        Leave department = new LeaveBuilder().build();
+
+        department.addEmployee(TypicalEmployees.ALICE);
+        assertTrue(department.hasEmployee(TypicalEmployees.ALICE));
+    }
+
+    @Test
+    public void execute_addDuplicateEmployee_throwsException() {
+        Leave department = new LeaveBuilder().build();
+
+        department.addEmployee(new EmployeeBuilder().build());
+
+        assertThrows(DuplicateEmployeeException.class, () -> department.addEmployee(new EmployeeBuilder().build()));
+    }
+
+    @Test
+    public void execute_addExactDuplicateEmployee_throwsException() {
+        Leave department = new LeaveBuilder().build();
+
+        department.addEmployee(TypicalEmployees.ALICE);
+
+        assertThrows(DuplicateEmployeeException.class, () -> department.addEmployee(TypicalEmployees.ALICE));
+    }
+
+    @Test
+    public void execute_removeEmployee_success() {
+        Leave leave = new LeaveBuilder().build();
+        leave.addEmployee(new EmployeeBuilder().build());
+
+        leave.deleteEmployee(new EmployeeBuilder().build());
+        assertTrue(leave.equals(new LeaveBuilder().build()));
+    }
+
+    @Test
+    public void execute_removeExactEmployee_success() {
+        Leave leave = new LeaveBuilder().build();
+        leave.addEmployee(TypicalEmployees.ALICE);
+
+        leave.deleteEmployee(TypicalEmployees.ALICE);
+        assertTrue(leave.equals(new LeaveBuilder().build()));
+    }
+
```

Comment:
What if you remove nonexistent employee?

Reply:
It does nothing as its implementation is similar to UniqueEmployeeList.remove()